.github/workflows/docs-preview.yml

```
@@ -0,0 +1,310 @@
+
+      - name: Post or update the preview comment
+        if: steps.eligibility.outputs.comment == 'true'
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+          HEAD_SHA: ${{ github.event.pull_request.head.sha }}
+          PUBLISHED: ${{ steps.eligibility.outputs.publish }}
+          PREVIEW_BASE_URL: ${{ vars.DOCS_PREVIEW_BASE_URL }}
+          ARTIFACT_NAME: ${{ needs.build-preview.outputs.artifact-name }}
+          RUN_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+        run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2016:info:12:12: Expressions don't expand in single quotes, use double quotes for that

(shellcheck)